src/border_agent/border_agent.cpp

```diff
@@ -115,6 +115,16 @@ void BorderAgent::Init(void)
 {
     mNcp.AddThreadStateChangedCallback([this](otChangedFlags aFlags) { HandleThreadStateChanged(aFlags); });
 
+#if OTBR_ENABLE_DBUS_SERVER
+    mNcp.GetThreadHelper()->SetUpdateMeshCopTxtHandler([this](std::map<std::string, std::vector<uint8_t>> aUpdate) {
+        this->mMeshCopTxtUpdate = std::move(aUpdate);
+        if (IsThreadStarted())
+        {
+            this->UpdateMeshCopService();
+        }
```

Comment:
should we also update meshcop service even when Thread is not started?

Reply:
We haven't implemented the feature to publish meshcop service when Thread is disabled.